store/secondary.go

```diff
@@ -0,0 +1,182 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"sync"
+
+	"github.com/Layr-Labs/eigenda-proxy/metrics"
+	"github.com/ethereum-optimism/optimism/op-service/retry"
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type MetricExpression = string
+
+const (
+	Miss    MetricExpression = "miss"
+	Success MetricExpression = "success"
+	Failed  MetricExpression = "failed"
+)
+
+type ISecondary interface {
+	AsyncEntry() bool
+	Enabled() bool
+	Topic() chan<- PutNotify
+	CachingEnabled() bool
+	FallbackEnabled() bool
+	HandleRedundantWrites(ctx context.Context, commitment []byte, value []byte) error
+	MultiSourceRead(context.Context, []byte, bool, func([]byte, []byte) error) ([]byte, error)
+	WriteSubscriptionLoop(ctx context.Context)
+}
```

Comment:
This probably works. I personally feel like we're getting wayyyy to complex way too early, for something that's super simple: we only have s3 and redis options, but now we have all these indirection layers, caching vs fallback, interfaces over interfaces. Makes it very hard to understand the code, review it, find bugs, benchmark/profile and optimize, etc. Let's merge this, but I'd like to try to keep the complexity lower in the future, or at least think through ways of simplifying the architecture.